cache/remotecache/v1/parse.go

```diff
@@ -82,7 +82,7 @@ func parseRecord(cc CacheConfig, idx int, provider DescriptorProvider, t solver.
 			}
 
 			remote.Descriptors = append(remote.Descriptors, descPair.Descriptor)
-			mp.Add(descPair.Descriptor.Digest, descPair.Provider)
+			mp.Add(descPair.Descriptor.Digest, descPair)
```

Comment:
Related to #4686 :

The original provider should be passed to the `multiprovider` without hiding methods with the wrapped struct (`DescriptorProviderPair`). Or, `DescriptorProviderPair` should export methods `multiprovider` relies on ( `UnlazySession` and `SnapshotLabels`). If this PR is still needed, I can work on this PR.

Reply:
Thanks! Feel free to carry this PR :)